pkgs/development/interpreters/octave/default.nix

```diff
@@ -59,6 +61,8 @@
 }:
 
 let
+  self = pkgs.octave;
```

Comment:
Although we wish to use `pkgs` which is usually discouraged, here I think you can use:

```nix
let
  octavePackages = import ..../octave-packages.nix {
    inherit self;
    # OR:
    octave = self;
    # ...
  };
  self = stdenv.mkDerivation {
    # ...
    passthru = {
      # ...
      pkgs = octavePackages;
      # ...
    };
    # ...
  };
in self
```

And you'll use `pkgs` a bit less extensively. I don't know if it's that important, it's a matter of personal style I think, so you don't have to do this. FYI there's https://github.com/NixOS/rfcs/pull/83 were we discuss a bit how to compose packages sets in general in Nixpkgs.